core/model/impl/src/main/kotlin/ConditionModelImpl.kt

```diff
@@ -174,13 +174,15 @@ internal class ConditionModelImpl private constructor(
 
         override fun toString(childContext: MayBeInvalid?) = buildRichString {
             color = TextColor.Red
-            // TODO: Maybe use `invalidExpression` here?
             append("<invalid-condition>")
         }
     }
 
     companion object Factory : BiObjectCache<Type, String, ConditionModelImpl>() {
-        operator fun invoke(type: Type, pathSource: String): ConditionModelImpl {
+        operator fun invoke(type: Type, pathSource: String): VariableBaseImpl {
+            if (!pathSource.matches(ConditionRegex)) {
+                return Invalid(type, pathSource)
```

Comment:
Got moved here (inside) from the `FeatureModelImpl`, along with `ConditionRegex`.